subgraph/src/fleek-nfa.ts

```diff
@@ -203,6 +206,41 @@ export function handleTokenRoleGranted(event: TokenRoleGrantedEvent): void {
   entity.transactionHash = event.transaction.hash;
 
   entity.save();
+
+  if (event.params.role === 1) {
+    // This is a new controller being added to a token.
+    // First we add the controller to the token's list of controllers.
+    // Then we create a new controller entity.
+
+    let token = Token.load(
+      Bytes.fromByteArray(Bytes.fromBigInt(event.params.tokenId))
+    );
+    let controller = Controller.load(event.params.toAddress);
+
+    if (!controller) {
+      // Create a new controller entity
+      log.debug('CONTROLLER IS GOING TO BE CREATED HERE.', []);
+      controller = new Controller(event.params.toAddress);
+    }
+
+    if (token !== null) {
+      let token_controllers = token.controllers;
+      if (!token_controllers) {
+        token_controllers = [];
+      }
+      token_controllers.push(event.params.toAddress);
```

Comment:
Don't we need to push a Controller object here instead of the raw address?

Reply:
No. Just the address works.